File: bpf_core.c
// bpf_core.c - Eve's framework functionality
//      includes pseudocode for planned functionality
// author - Garrett Fechter garrett.fechter@gmail.com
//

#include <linux/if_ether.h>
#include <linux/if_packet.h>
#include <linux/ip.h>
#include <uapi/linux/pkt_cls.h>
#include <bcc/proto.h>
#include <net/sock.h>

// need to decide how to forward packets from one interface to another
// tentative: tc bpf using bpf_redirect
//
// hash1 is incoming packet data, hash1[0] is current index
//
// if (changingPacket)
//      serial.light3.blink()
// if (packet.isIncoming())
//      cyclicalIndex = hash1[0]
//      hash1[cyclicalIndex] = packet
//      cyclicalIndex++
//      if (cyclicalIndex > hash1.len)
//           cyclicalIndex = 1

File: bpf_core.py
# bpf_core.py - Eve's framework functionality
#   includes pseudocode for planned functionality
# author - Garrett Fechter garrett.fechter@gmail.com

from __future__ import print_function
from bcc import BPF
from pyroute2 import IPRoute
import time
import socket

# b = BPF(src_file="bpf_core.c")

# while 1
# sleep 1
# index = hash1[0]
# if index > prev_index
#   save hash1[prev_index] to hash1[index] on USB
# elif index < prev_index
#   save hash1[prev_index] to hash1[max]
#   save hash1[1] to hash1[index]
